mhw_library/measures.py

```diff
@@ -0,0 +1,605 @@
+    
+    Returns
+    -------
+    np.ndarray
+        An array of the deformation of the object at each timestamp.
+    
+    Notes
+    -----
+    This function is my implementation of a measure that calculates the deformation,
+    which is the complement to the overlap or shared area between consecutive timestamps.
+    The calculation is based on the algorithm presented in Sun, D., Jing, Z., Li, F., 
+    Wu, L., "Characterizing Global Marine Heatwaves Under a Spatio-temporal Framework," 
+    Progress in Oceanography (2022).
+    """
+    return np.asarray(1 - np.asarray(sharedarea_ls))
+
+def calc_largest_and_smallest_timestamp(spatial_extents: List[float]) -> Tuple[float, float]:
+    """
+    Calculates the timestamp at which the object has the largest and smallest spatial extents.
+    
+    Parameters
+    ----------
+    spatial_extents : list of floats
+        The spatial extents of the object at each timestamp.
+
+    Returns
+    -------
+    Tuple[float, float]
+        A tuple containing the percentage of the time series at which the object has its 
+        largest and smallest spatial extents, respectively.
+    """
+    when_large = np.argmax(spatial_extents) 
+    when_small = np.argmin(spatial_extents)
+    return when_large, when_small
+
+def calc_spatial_cross_correlation(merged_xarray: xr.Dataset, mhw_id: int, duration: int) -> np.ndarray:
+    """
+    Calculates the spatial cross correlation of an object in the merged_xarray dataset
+
+    Parameters
+    ----------
+    merged_xarray : xr.Dataset
+        A dataset containing the merged Sea Surface Temperature Anomaly (SSTA) data and labels
+    mhw_id : int
+        The identifier of the object to compute the spatial cross correlation for
+    duration : int
+        The number of timesteps where the object with the specified 
+        identifier appears in the merged_xarray.
+    Returns
+    -------
+    np.ndarray
+        An array containing the spatial cross correlation values for the object at each timestep
+    """
+    one_obj = merged_xarray.where(merged_xarray.labels==mhw_id, drop=False)
+    first_timestep, first_array, month = utils.calc_initialization(merged_xarray, mhw_id)
+    timesteps_to_choose_from = np.arange(first_timestep, first_timestep+duration)
+    cc_image_array = np.zeros((len(timesteps_to_choose_from), 192,288))    
+    
+    for i in range(len(timesteps_to_choose_from[:-1])):
+        image = one_obj.SSTA[timesteps_to_choose_from[i],:,:].values
+        image = np.nan_to_num(image)
+        offset_image = one_obj.SSTA[timesteps_to_choose_from[i+1],:,:].values
+        offset_image = np.nan_to_num(offset_image)
+        image_product = np.fft.fft2(image) * np.fft.fft2(offset_image).conj()
+        cc_image = np.fft.fftshift(np.fft.ifft2(image_product))
+        cc_image_array[i,:,:] = np.real(cc_image)
+    
+    return cc_image_array
+
+def calc_ratio_convexhullarea_vs_objarea(merged_xarray: xr.Dataset, mhw_id: int) -> list:
```

Comment:
maybe just calc_ratio_convexhullarea_vs_area
since both the convexhullarea and the area correspond to the same object.